device_backends/pcie/src/PcieBackend.cc

```diff
@@ -126,10 +134,7 @@ namespace ChimeraTK {
     l_RW.size_rw = 0; // does not overwrite the struct but writes one word back to data
     l_RW.data_rw = -1;
     l_RW.rsrvd_rw = 0;
-    if(::read(_deviceID, &l_RW, sizeof(device_rw)) != sizeof(device_rw)) {
-      return false;
-    }
-    return true;
+    return ::read(_deviceID, &l_RW, sizeof(device_rw)) == sizeof(device_rw);
```

Comment:
unrelated, but is this always an all-or-nothing read, Can it never be partial? Or do we handle partial reads as fail?